### PR TITLE
prefer local over global npmrc file

### PR DESCRIPTION
**Pull request by svnlto**
This PR changes the default behaviour of this library so it'll look for a local `npmrc` file first before defaulting to global.

I found this to be very useful in build scenarios that are not able to provide encapsulated environments.

### CODE REVIEW

File: .npmrc [hunk 1]
```diff
@@ -1,2 +1,3 @@
 registry=http://registry.npmjs.org/
 save-exact=true
+//npm.example.com/:_authToken=${NPM_TOKEN}
```
Comment (bahmutov) [inline .npmrc]: could you remove this line - break CI

**Comment by bahmutov** [comment]
Love it! Could you remove the token in the `.npmrc` file and push the change, and I will merge it

**Comment by svnlto** [comment]
Done 👍

**Comment by svnlto** [comment]
Looks like something else is breaking now?

**Comment by bahmutov** [comment]
yeah, seems one of the tests it putting NPM_TOKEN into the `.npmrc` and then trying to run more tests fails. I will merge your pull request and will fix this later tonight.

Thank you for the code, will try to publish tonight

**Comment by hbetts** [comment]
@bahmutov I recently came across this pull request, because the functionality described by @svnlto is exactly what I'm looking for to help with isolating unit tests for `npm-publish-git-tags` - https://gitlab.com/hyper-expanse/npm-publish-git-tag/merge_requests/2/

I'd just like to ask, when you have a few minutes, if you wouldn't mind publishing this pull request to the npm registry so I can use it in my project.

**Comment by bahmutov** [comment]
Version `1.13.0` has been published to NPM. The full release note can be found at [npm-utils/releases/tag/v1.13.0](https://github.com/bahmutov/npm-utils/releases/tag/v1.13.0).

**Tip:** safely upgrade dependency npm-utils in your project using [next-update](https://github.com/bahmutov/next-update)

**Comment by bahmutov** [comment]
@hbetts I have pushed this feature (BTW I implemented https://github.com/bahmutov/github-post-release based on your https://github.com/gitter-badger/semantic-release-github-notifier) and was released.

**Pull request merged**